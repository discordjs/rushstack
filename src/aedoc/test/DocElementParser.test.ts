// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import { DocElementParser } from '../DocElementParser';

function makeSnapshot(s: string): void {
  expect(DocElementParser.parseMarkdownishText(s)).toMatchSnapshot();
}

test('DocElementParser.parseMarkdownishText', () => {
  makeSnapshot('not an escape:  C:\\Dog\\Cat');
  makeSnapshot('escaped symbols: \\$\\^');
  makeSnapshot('escaped backslashes: \\\\ \\\\\\ end');
  makeSnapshot('escaped HTML: \\<td>');
  makeSnapshot('real HTML: <td> </td> <a href="#" /a>');
});
