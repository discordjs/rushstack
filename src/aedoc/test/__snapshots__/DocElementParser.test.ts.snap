// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocElementParser.parseMarkdownishText 1`] = `
Array [
  Object {
    "kind": "text",
    "text": "not an escape: C:\\\\Dog\\\\Cat",
  },
]
`;

exports[`DocElementParser.parseMarkdownishText 2`] = `
Array [
  Object {
    "kind": "text",
    "text": "escaped symbols: $^",
  },
]
`;

exports[`DocElementParser.parseMarkdownishText 3`] = `
Array [
  Object {
    "kind": "text",
    "text": "escaped backslashes: \\\\ \\\\\\\\ end",
  },
]
`;

exports[`DocElementParser.parseMarkdownishText 4`] = `
Array [
  Object {
    "kind": "text",
    "text": "escaped HTML: <td>",
  },
]
`;

exports[`DocElementParser.parseMarkdownishText 5`] = `
Array [
  Object {
    "kind": "text",
    "text": "real HTML: ",
  },
  Object {
    "kind": "html-tag",
    "token": "<td>",
  },
  Object {
    "kind": "text",
    "text": " ",
  },
  Object {
    "kind": "html-tag",
    "token": "</td>",
  },
  Object {
    "kind": "text",
    "text": " <a href=\\"#\\" /a>",
  },
]
`;
